crates/core/tedge_agent/src/entity_manager/tests.rs

```diff
@@ -0,0 +1,655 @@
+use crate::entity_manager::server::EntityStoreResponse;
+use crate::entity_manager::tests::model::Action;
+use crate::entity_manager::tests::model::Command;
+use crate::entity_manager::tests::model::Commands;
+use crate::entity_manager::tests::model::Protocol::MQTT;
+use proptest::proptest;
+use std::collections::HashSet;
+use tedge_actors::Server;
+use tedge_api::entity::EntityMetadata;
+
+#[tokio::test]
+async fn new_entity_store() {
+    let (mut entity_store, _mqtt_output) = entity::server("device-under-test");
+
+    assert_eq!(
+        entity::get(&mut entity_store, "device/main//").await,
+        Some(EntityMetadata::main_device())
+    )
+}
+
+#[tokio::test]
+async fn removing_an_unknown_child_using_mqtt() {
+    let registrations = vec![
+        // tedge mqtt pub -r te/device/a// ''
+        Command {
+            protocol: MQTT,
+            action: Action::RemDevice {
+                topic: "a".to_string(),
+            },
+        },
+    ];
+    check_registrations(Commands(registrations)).await
+}
+
+proptest! {
+    //#![proptest_config(proptest::prelude::ProptestConfig::with_cases(1000))]
+    #[test]
+    fn it_works_for_any_registration_order(registrations in model::walk(10)) {
+        tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(check_registrations(registrations))
```

Comment:
Extracting the check from the proptest helps to replay issues found with proptest.